Libraries/src/Amazon.Lambda.Annotations/README.md

```diff
@@ -53,9 +68,113 @@ public class Functions
 }
 ```
 
-## Using References To Other Resources and Parameters in the template
+Lambda Annotations uses C# source generators to generate that boiler plate code to brige the gap between the default Lambda programming model to Lambda Annotations programming model at compile time.
+In addition the source generator also synchonizes the CloudFormation template to declare all of the .NET methods with the `LambdaFunction` attribute as 
+Lambda functions in the CloudFormation template.
+
+## Dependency Injection integration
+
+Lambda Annotations supports dependency injection. A class can be marked with a `LambdaStartup` attribute. The class will 
+have a `ConfigureServices` method for configuring services.
+
+The services can be injected by either constructor injection or using the `FromServices` attribute on a method parameter of
+the function decorated with the `LambdaFunction` attribute.
+
+Services injected via the constructor have a lifecycle for the length of the Lambda compute container. For each Lambda 
+invocation a scope is created and the services injected using the `FromServices` attribute are created within the scope.
+
+Example startup class:
+```csharp
+[LambdaStartup]
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddAWSService<Amazon.S3.IAmazonS3>();
+        services.AddScoped<ITracker, DefaultTracker>();
+    }
+}
+```
+
+Example function using DI:
+```csharp
+public class Functions
+{
+    IAmazonS3 _s3Client;
+
+    public Functions(IAmazonS3 s3Client)
+    {
+        _s3Client = s3Client;
+    }
+
+
+    [LambdaFunction]
+    [HttpApi(LambdaHttpMethod.Put, "/process/{name}")]
+    public async Task Process([FromServices] ITracker tracker, string name, [FromBody] string data)
+    {
+        tracker.Record();
+
+        await _s3Client.PutObjectAsync(new PutObjectRequest
+        {
+            BucketName = "storage-bucket",
+            Key = name,
+            ContentBody = data
+        });
+    }
+}
+```
+
+## Synchronizing CloudFormation template
+
+When the .NET project is compiled the Lambda Annotation source generator will synchronize all of the C# methods with the `LambdaFunction` attribute in the 
+project's CloudFormation template. Support is available for both JSON and YAML based CloudFormation templates.
+The source generator identifies the CloudFormation template for the project by looking at the `template` property in the `aws-lambda-tools-defaults.json` 
+file. If either the property or file does not exist the source generator defaults to `serverless.template` and creates the file if it doesn't exist.
```

Comment:
Could we reword this to something like "If the 'template' property is absent, the source generator will default to the 'serverless.template' and create the file if it does not exist.

Reply:
Done